components/invitee.jsx

```diff
@@ -2,60 +2,114 @@ import React, { useState, useEffect } from 'react';
-                        <th>Date/time at scheduling</th>
-                        <th>Questions + Answers</th>
-                        <th>Rescheduled</th>
-                        <th>Timezone</th>
-                    </tr>
-                </thead>
-                <tbody>
-                    {invitees.map((invitee, i) => (
-                        <tr key={i}>
-                            <td>{invitee.name}</td>
-                            <td>{invitee.email}</td>
-                            <td>{invitee.scheduled_at}</td>
-                            <td>{invitee.questions_and_answers && invitee.questions_and_answers.length ? invitee.questions_and_answers.map((questAns) => (
-                                <ul key={questAns.position}>
-                                    <li>{questAns.position + 1}. <strong>Question: </strong> {questAns.question} <br />
-                                        <strong>Answer: </strong> {questAns.answer}</li>
-                                </ul>
-                            )) : 'N/A'}</td>
-                            <td>{invitee.rescheduled === false ? 'No' : 'Yes'}</td>
-                            <td>{invitee.timezone}</td>
-                        </tr>
-                    ))}
-                </tbody>
-            </table>
-        </div>
-    )
-}
+  const { uuid } = useParams();
+
+  const fetchData = async () => {
+    const nextPageQueryParams = pagination.next_page
+      ? pagination.next_page.slice(pagination.next_page.indexOf('?'))
+      : '';
+
+    const result = await fetch(
+      `/api/events/${uuid}/invitees${nextPageQueryParams}`
+    ).then((res) => res.json());
+
+    setInvitees([...result.invitees]);
+    setPagination(result.pagination);
+  };
+
+  const handleNoShowClick = async (event) => {
+    event.preventDefault();
+
+    const body = await JSON.stringify({ invitee: event.target.value });
+
+    await fetch('/api/no_shows', {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: body,
+    });
+
+    fetchData();
+  };
+
+  const undoNoShowClick = async (event) => {
+    event.preventDefault();
+
+    const filteredInvitees = invitees.filter(
+      (invitee) => invitee.uri === event.target.value
+    );
+
+    await fetch(
+      `/api/no_shows/${filteredInvitees[0].no_show.uri.split('/')[4]}`,
+      {
+        method: 'DELETE',
+      }
+    );
+
+    fetchData();
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  return (
+    <div>
+      <table className="striped centered">
+        <thead>
+          <tr>
+            <th>Name</th>
+            <th>E-mail</th>
+            <th>Date/time at scheduling</th>
+            <th>Questions + Answers</th>
+            <th>Rescheduled</th>
+            <th>Timezone</th>
+          </tr>
+        </thead>
+        <tbody>
+          {invitees.map((invitee, i) => (
+            <tr key={i}>
```

Comment:
It's not ideal to use an index as `key` w/ React. See [here](https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318). I would use something else unique like the `uri`.

Reply:
Indeed. I meant to change this! Thank you for pointing this out.